### DISCUSSION TITLE
Showing Progress across Multiple Components

### DISCUSSION THREAD

**Discussion by niallermoran** [vercel/next.js#63176] [Help]
### Summary

I am looking for guidance on how to display loading progress on multiple components when an action is executed on one. 

### Additional information

```js
My page is made up of a filter component and multiple other components that display data, charts, etc. I am using the App router

The filter component contains a form that uses a server action to update cookies with updated values (think global settings). 

The page then refreshes and components update within a second or two.

However, I cannot get any loading indicator to work. I have wrapped each component in suspense, which only works when page is navigated to. I have also tried just using client components and useState to manage an isLoading property, but that value is saved across the reset.

Their must be a pattern to solve this.

Can anyone help.

Thank you in advance


 I can call useRouter().refresh on al
```

**Comment by icyJoseph** [comment]
Hi,

You could try something very cheap, and add a key to the suspense boundaries, if possible, something related to the current state of the app. For example, the URL search params and such. Does that do the trick? I am a bit short on time, and can't really parse all of your input, but from what I quickly read, you already have suspense boundaries applied, so it might just work.

**Reply by niallermoran** [reply]
Hey there. Gave it a try but no luck. Just so I am clear did you mean add a key attribute to the suspense component around my components? 

Thanks for helping

**Reply by icyJoseph** [reply]
Yeah, and that should change based on the current data you are showing. 

I think maybe this issue comment explains it better:

- https://github.com/vercel/next.js/issues/53543#issuecomment-1664793532
